web/src/components/Korrel8rPanel.tsx

```diff
@@ -27,46 +27,60 @@ import { TFunction, useTranslation } from 'react-i18next';
 import { CubesIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
 import { useURLState } from '../hooks/useURLState';
 import { usePluginAvailable } from '../hooks/usePluginAvailable';
+import { useDispatch, useSelector } from 'react-redux';
+import { State } from '../redux-reducers';
+import { Query, QueryType, setPersistedQuery } from '../redux-actions';
 
 type Result = {
   graph?: Korrel8rGraphResponse;
   message?: string;
   title?: string;
 };
-enum QueryType {
-  Neighbour,
-  Goal,
-}
 
 export default function Korrel8rPanel() {
   const { t } = useTranslation('plugin__troubleshooting-panel-console-plugin');
+  const persistedQuery = useSelector((state: State) => {
+    return state.plugins?.tp?.get('persistedQuery');
+  }) as Query;
+  const dispatch = useDispatch();
 
   // State
   const { korrel8rQueryFromURL } = useURLState();
-  const [query, setQuery] = React.useState(korrel8rQueryFromURL); // Initial value from URL
+  // Initial value from persisted query if available (when opened from anywhere but
+  // the alerts page or if never opened before), otherwise URL
+  const initialQuery = persistedQuery.query
+    ? persistedQuery
+    : {
+        query: korrel8rQueryFromURL,
+        queryType: QueryType.Neighbour,
+        depth: 3,
+        goal: null,
+      };
+  const [query, setQuery] = React.useState<Query>(initialQuery);
   const [result, setResult] = React.useState<Result | null>(null);
   const [showQuery, setShowQuery] = React.useState(false);
-  const [queryType, setQueryType] = React.useState(QueryType.Neighbour);
-  const [depth, setDepth] = React.useState(3);
-  const [goal, setGoal] = React.useState('');
 
   React.useEffect(() => {
     // Set result = null to trigger a reload, don't run the query till then.
     if (result !== null) {
       return;
     }
     let fetch: CancellableFetch<Korrel8rGraphResponse>;
-    if (queryType === QueryType.Neighbour) {
-      fetch = getNeighborsGraph({ query }, depth);
-    } else if (queryType === QueryType.Goal) {
-      fetch = getGoalsGraph({ query }, goal);
+    if (query.queryType === QueryType.Neighbour) {
+      fetch = getNeighborsGraph(query);
+    } else if (query.queryType === QueryType.Goal) {
+      fetch = getGoalsGraph(query);
     } else {
       return;
     }
     const { request, abort } = fetch;
     request()
       .then((response: Korrel8rGraphResponse) => {
         setResult({ graph: { nodes: response.nodes, edges: response.edges } });
+        // Only set the persisted query upon a successful query. It would be a
```

Comment:
I can think of cases where we want to save a failed query:
- the _user_ has constructed a bad query which is a "work in progress" they want to edit.
- query fails for transient reasons (missing plugin, problem talking to store...) and may pass later.

The trouble is that our "open the panel" button is ambiguous - does it mean "take up where I left off" or "focus on what I'm currently looking at". Since we can't tell the difference, we have to assume "take up where I left off" and leave the user with a single click on "Focus" if that's the wrong assumption. 
This seems OK to me for extended preview. In the long run I think we should agree a common approach for both troubleshooting and lightspeed. Either way the save/restore code for panel state will be still be relevant.